### ISSUE TITLE
block notifications during night

### ISSUE THREAD

**Issue by FransM** [Freeyourgadget/Gadgetbridge#1706]
#### Before requesting a new feature, please confirm the following:
- [ x] I have read the [wiki](https://github.com/Freeyourgadget/Gadgetbridge/wiki), and I didn't find a solution to my problem / an answer to my question.
- [x ] I have searched the [issues](https://github.com/Freeyourgadget/Gadgetbridge/issues), and I didn't find a solution to my problem / an answer to my question.
- [ x] If you upload an image or other content, please make sure you have read and understood the [github policies and terms of services](https://help.github.com/articles/github-terms-of-service/#1-responsibility-for-user-generated-content)

#### Your issue is:

This is somewhat similar to #1255 but I decided to write a separate ticket for a more general request.

What I would like to have is to disable sending notifications to the band/watch during night.
One option could be to disable all notifications if the screen is set to dim.
The other more flexible solution would be to allow specifying the time for this feature.

Ideally this would be a per day setting so one can select a different range per week day (so e.g. no notifications between 23.00 and 07.00 during weekdays and not between 23.30 and 08.00 on saturdays and not between 23.30 and 09.00 on sundays

#### Your wearable device is:

mi band 4

#### Your android version is:
7.1

#### Your Gadgetbridge version is:
latest from fdroid

*New requests about already solved/documented topics could be closed without further comments. Same for too generic or incomplete reports.*

**Comment by newhinton** [comment]
I think this would be a good idea, however i dont think this should be implemented via time. I would suggest to get the phonestate, volumewise, and only send notifications if the device is either loud or on vibrate. If the device is set to muted, gadgetbridge should not send a notification. (If that feature is enabled)

I was thinking about implementing such behaviour

**Comment by water1309** [comment]
Why "shouldn't" this be triggered via time? My phone is always on vibrate for calls and silent for anything else (and some notifications I get on my mi band). So your suggested implementation would not be useful in my situation.

Wouldn't a time based trigger be easier to  implement? Maybe both triggers could be implemented.

**Comment by newhinton** [comment]
I agree, it would not help in your situation. It should not be hard to implement both. However, timebased switching as you suggest is obviously more work from the ui perspective. 

Assuming this feature was already implemented, what ui would you expect? A simple "from - to" , or a more elaborate solution? 

Also since i am not a maintainer, input from one (especially regarding the ui) would be highly welcomed :D

**Comment by newhinton** [comment]
@water1309 @FransM I was thinking about the user interface. We could use a list to define times where the device should discard notifications, which would be quite much work in the UI (in comparison to my next idea)

Wouldn't it be also possible to simply select a device-calendar on which the device is ignoring notifications? That reduces the work on the ui a bit and in combination with ignoring the notification when the device is silent it should capture most usecases.

**Comment by cpfeiffer** [comment]
In my opinion, we could borrow the UI from Android's DND "automatic rules". You can add rules for any day combination. All rules have a start and end time. We don't need the calendar-based rules, I think.

These settings should be put into a small settings fragment that would also host the Do not disturb setting (in the settings below the other notification options).

**Comment by FransM** [comment]
As mentioned in the ticket a very simple solution would be to not forward notifications during the time the band is programmed to dim. Guess the only UI change would be to enable/disable this

**Comment by newhinton** [comment]
@FransM I dont really like that solution, because it does not nessessarily match the wanted timeframe. Also daylightsavings can make it behave strangely and a little bit "unpredictable".

To me, Time and Volume looks like a more straightforward approach

**Comment by newhinton** [comment]
@cpfeiffer Yesterday i found out the hard way that sms does not count as a notification (and was therefore not blocked).

It seems only NotificationListener has a dedicated shouldIgnore function, and every external event has it's own rules for letting things through or not.

While that makes sense sometimes (eg. i do not need to check if a phonenumber is ignored when i recieved an ad from an app), sometimes things apply generally (like the dnd setting which i tried to implement. No calls for me at night! :D )

How should i proceed now? Should i try to build something more general which can block all kinds of notifications, or just make this work for each case on it's own?

Also, what's up with the migration, what do i need to do? (do i need to do anything except switching remotes?)
